src/pydvl/value/oob/oob.py

```diff
@@ -0,0 +1,142 @@
+    \mathbb{1}
+    (w_{bi}=0)}$$
+
+    With:
+
+    $$
+    T: Y \times Y
+    \rightarrow \mathbb{R}
+    $$
+
+    T is a score function that represents the goodness of a weak learner $\hat{f}_b$ at the i-th datum $(x_i, y_i)$.
+
+    There is a need to tune n_est and max_samples jointly to ensure all samples are at least 1 time oob, otherwise the result could include a nan value for that datum.
+
+    Args:
+        u: Utility object with model, data, and scoring function.
+        n_est: Number of estimator used in the bagging procedure.
+        max_samples: The fraction of samples to draw to train each base estimator.
+        n_jobs: The number of jobs to run in parallel used in the bagging procedure for both fit and predict.
+        loss: A function taking as parameters model prediction and corresponding data labels(preds, y) and returning an array of point wise errors.
+        progress: If True, display a progress bar.
+
+    Returns:
+        Object with the data values.
+    """
+
+    result = ValuationResult.empty(
+        algorithm="data_oob", indices=u.data.indices, data_names=u.data.data_names
+    )
+    # check model type:
+    if is_classifier(u.model):
+        model_type = "classifier"
+    elif is_regressor(u.model):
+        model_type = "regressor"
+    else:
+        raise Exception(
+            "Model has to be classifier or a regressor in a sklearn format."
+        )
+
+    if model_type == "classifier":
+        bag = BaggingClassifier(
+            u.model, n_estimators=n_est, max_samples=max_samples, n_jobs=n_jobs
+        )
+        if loss is None:
+            loss = point_wise_accuracy
+    else:
+        bag = BaggingRegressor(
+            u.model, n_estimators=n_est, max_samples=max_samples, n_jobs=n_jobs
+        )
+        if loss is None:
+            loss = neg_l2_distance
+
+    bag.fit(u.data.x_train, u.data.y_train)
+
+    for est, samples in maybe_progress(
+        zip(bag.estimators_, bag.estimators_samples_), progress, total=n_est
+    ):  # The bottleneck is the bag fitting not this part so TQDM is not very useful here
+        oob_idx = np.setxor1d(u.data.indices, np.unique(samples))
+        array_loss = loss(
+            preds=est.predict(u.data.x_train[oob_idx]), y=u.data.y_train[oob_idx]
+        )
+        result += ValuationResult(
+            algorithm="data_oob",
+            indices=oob_idx,
+            values=array_loss,
+            counts=np.ones_like(array_loss, dtype=u.data.indices.dtype),
+        )
+    return result
+
+
+def point_wise_accuracy(preds: NDArray, y: NDArray) -> NDArray:
+    r"""Computes point wise accuracy
+
+    Args:
+        preds: Model prediction on
+        y:  data labels corresponding to the model predictions
+
+    Returns:
+        Array of point wise accuracy
+    """
+    return np.array(
+        preds == y,
+        dtype=bool,
+    )
+
+
+def neg_l2_distance(preds: NDArray, y: NDArray) -> NDArray[np.float64]:
```

Comment:
This function is actually generic. With 

```python
T = TypeVar("T", bound=np.number)
```
you can do

```suggestion
def neg_l2_distance(preds: NDArray[T], y: NDArray[T]) -> NDArray[T]:
```

and then use `preds.dtype` as the return value's `dtype`.